### PR TITLE
improved uri handling on windows

### PR DESCRIPTION
**Pull request by cdietrich**
This fixes the uri en- and decoding problems on windows.

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by cdietrich** [comment]
@svenefftinge @spoenemann can you have a look

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/IFileSystemScanner.xtend [hunk 1]
```diff
@@ -29,13 +29,19 @@ interface IFileSystemScanner {
...
+            acceptor.accept(uri)
             if (file.isDirectory) {
                 for (f : file.listFiles) {
                     scanRec(f, acceptor)
                 }
             }
         }
         
+        def String toPath(java.net.URI uri) {
+            val path = Paths.get(uri)
```
Comment (svenefftinge) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/IFileSystemScanner.xtend]: What is this conversion doing?

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/IFileSystemScanner.xtend [hunk 2]
```diff
@@ -29,13 +29,19 @@ interface IFileSystemScanner {
...
+            val uri = URI.createURI(file.absoluteFile.toURI.toPath)
+            acceptor.accept(uri)
             if (file.isDirectory) {
                 for (f : file.listFiles) {
                     scanRec(f, acceptor)
                 }
             }
         }
         
+        def String toPath(java.net.URI uri) {
```
Comment (svenefftinge) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/IFileSystemScanner.xtend]: the name 'toPath' is missleading if it returns a string representation of a URI.

**Comment by cdietrich** [comment]
@svenefftinge i added some comments why we need this convertion and inlined the method.
btw the very same toPath naming is inside UriExtension

**Comment by svenefftinge** [comment]
It's a bad name there, too. Maybe we should move a well named implementation to UriUtil?

**Comment by cdietrich** [comment]
which name would you prefer? can you file a separate ticket for that?

**Comment by svenefftinge** [comment]
+1 on merging this

**Pull request merged**